### PR TITLE
Gary loop

### PR DESCRIPTION
**Pull request by cheungaryk**
## loops
added [loops](https://www.hackerrank.com/challenges/python-loops/problem)

## misc
- add code coverage to unit test
- add [virtual environment](https://docs.python.org/3/library/venv.html) files for pytest and pytest-cov modules
- add [Makefile](https://opensource.com/article/18/8/what-how-makefile) for easier test command. Try `make test` (you may need to activate the virtual environment above first)
- minor refactor and clean up

### CODE REVIEW

File: template/python/4_loops/loops.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/python3
+#
+# https://www.hackerrank.com/challenges/python-arithmetic-operators/problem
```
Comment (brdgcn) [inline template/python/4_loops/loops.py]: I don't think this link is correct

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+PHONY: test
+pytest test:
+	pytest --cov=./
```
Comment (brdgcn) [inline Makefile:3]: is py.test same as pytest?

File: gary/python/3_arithmetic_operators/arithmetic_operators_test.py [hunk 3]
```diff
@@ -4,47 +4,90 @@
...
-        self.assertEqual(result, 2)
-    
-    def test_minus_2(self):
-        result = minus(666, 43)
-        self.assertEqual(result, 623)
 
-    def test_minus_3(self):
-        result = minus(1, 2)
-        self.assertEqual(result, -1)
+    def minus_base_test(self, num1, num2, expected_result):
```
Comment (brdgcn) [inline gary/python/3_arithmetic_operators/arithmetic_operators_test.py:38]: should def  minus_base_test be
def test_minus_base
otherwise it gets skipped?

File: Makefile (hunk 2 above)
Comment (cheungaryk) [reply Makefile:3]: there is no py.test, only pytest. This syntax allows one to enter either `make pytest` or `make test` commands to run pytest with coverage

File: template/python/4_loops/loops.py (hunk 1 above)
Comment (cheungaryk) [reply template/python/4_loops/loops.py]: this has been fixed. Good catch

File: gary/python/3_arithmetic_operators/arithmetic_operators_test.py (hunk 3 above)
Comment (cheungaryk) [reply gary/python/3_arithmetic_operators/arithmetic_operators_test.py:38]: I want it to be skipped because this method's purpose is to be reused by other methods

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -18,14 +18,13 @@ jobs:
...
-# pip install -r python/requirements.txt
     - name: Lint with flake8
       run: |
         pip install flake8
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+        # the GitHub editor is 127 chars wide
+        flake8 . --count --max-complexity=10 --max-line-length=127 --statistics
```
Comment (brdgcn) [inline .github/workflows/tests.yml:27]: by deleting exit-zero, errors are no longer warnings and will return an error message, correct? 
curious why the change in this test file.

File: gary/python/3_arithmetic_operators/arithmetic_operators_test.py [hunk 5]
```diff
@@ -4,47 +4,90 @@
...
-    def test_add_2(self):
-        result = add(666, 43)
-        self.assertEqual(result, 709)
-    
-    def test_add_3(self):
-        result = add(1, 2)
-        self.assertEqual(result, 3)
+    def add_base_test(self, num1, num2, expected_result):
+        result = add(num1, num2)
+        self.assertEqual(result, expected_result)
```
Comment (brdgcn) [inline gary/python/3_arithmetic_operators/arithmetic_operators_test.py:12]: wow, such an efficient way to write multiple assertEqual methods(functions?)!

**Review by brdgcn — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 4 above)
Comment (cheungaryk) [reply .github/workflows/tests.yml:27]: yes, I want to make the test fail when linting returns errors, so we can follow the lints correctly
